plugins/TapTempo/TapTempo.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * TapTempo.cpp - Plugin to count beats per minute
+ *
+ *
+ * Copyright (c) 2022 saker <sakertooth@gmail.com>
+ *
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "TapTempo.h"
+
+#include <QFont>
+#include <QKeyEvent>
+#include <QLabel>
+#include <QPushButton>
+#include <QVBoxLayout>
+#include <cfloat>
+#include <cmath>
+#include <string>
+
+#include "AudioEngine.h"
+#include "Engine.h"
+#include "LedCheckBox.h"
+#include "SamplePlayHandle.h"
+#include "Song.h"
+#include "embed.h"
+#include "plugin_export.h"
+
+namespace lmms {
+
+extern "C" {
+Plugin::Descriptor PLUGIN_EXPORT taptempo_plugin_descriptor
+	= {LMMS_STRINGIFY(PLUGIN_NAME), "Tap Tempo", QT_TRANSLATE_NOOP("PluginBrowser", "Tap to the beat"),
+		"saker <sakertooth@gmail.com>", 0x0100, Plugin::Tool, new PluginPixmapLoader("logo"), nullptr, nullptr};
+
+TapTempo::TapTempo()
```

Comment:
It looks like `TapTempo::TapTempo()` and `void TapTempo::onBpmClick()` are defined within the `extern "C"` block when they should be outside it. You could probably just move `lmms_plugin_main(...)` up to the top so that it can be grouped together with the descriptor, then the `TapTempo` methods could come after that. However you want to do it

Reply:
Yeah, I must've put everything into ``extern "C"`` by accident. Nice catch :+1: